### PR TITLE
Add ability to close PMs

### PR DESCRIPTION
**Pull request by nekodex**
Leaving/closing a PM channel will hide it until a new message arrives in the channel.

Messaging the same user again will show the past chat history (if present and not culled), although right now the experience of re-opening a chat is a bit awkward and requires sending a message before the history is reloaded/shown (will fix later™).

### CODE REVIEW

File: resources/assets/lib/stores/channel-store.ts [hunk 1]
```diff
@@ -107,6 +114,20 @@ export default class ChannelStore implements DispatchListener {
     });
   }
 
+  @computed
+  get channelList(): Channel[] {
+    let channels: Channel[] = [];
+    channels = channels.concat(this.nonPmChannels);
+    channels = channels.concat(this.pmChannels);
+
+    return channels;
```
Comment (nanaya) [inline resources/assets/lib/stores/channel-store.ts]: Just one line

    return [...this.a, ...this.b]

?

File: resources/assets/lib/chat/conversation-list-item.tsx [hunk 2]
```diff
@@ -33,33 +33,45 @@ export default class ConversationListItem extends React.Component<any, {}> {
...
-      <a href='#' className={className} onClick={this.switch}>
-        <div className='chat-conversation-list-item__unread-indicator' />
-        <img className='chat-conversation-list-item__avatar' src={conversation.icon} />
-        <div className='chat-conversation-list-item__name'>{conversation.name}</div>
-        <div className='chat-conversation-list-item__chevron'>
-          <i className='fas fa-chevron-right' />
-        </div>
-      </a>
+      <div className={className}>
+        <a href='#' className={`${baseClassName}__close-button`} onClick={this.part}>
```
Comment (nanaya) [inline resources/assets/lib/chat/conversation-list-item.tsx]: is there reason this isn't button? (so `preventDefault` isn't needed)

File: resources/assets/lib/chat/chat-api.ts [hunk 3]
```diff
@@ -68,6 +68,21 @@ export default class ChatAPI {
     });
   }
 
+  partChannel(channelId: number, userId: number) {
+    return new Promise((resolve, reject) => {
```
Comment (nanaya) [inline resources/assets/lib/chat/chat-api.ts]: `$.ajax` has `.then` and `.catch` since 3.0 so the wrapping isn't needed anymore.

Comment (nekodex) [reply resources/assets/lib/chat/chat-api.ts]: noted

**Pull request merged**